### PR TITLE
Update OpenTelemetry Dependencies to latest

### PR DESCRIPTION
**Pull request by srv-gh-o11y-gdi**
Update OpenTelemetry Dependencies to latest

### CODE REVIEW

File: internal/receiver/signalfxgatewayprometheusremotewritereceiver/server.go [hunk 1]
```diff
@@ -38,13 +38,13 @@ type prometheusRemoteWriteServer struct {
...
 
 type serverConfig struct {
-	confighttp.ServerConfig
 	component.TelemetrySettings
 	Reporter reporter
 	component.Host
 	Mc     chan<- pmetric.Metrics
 	Parser *prometheusRemoteOtelParser
 	Path   string
+	confighttp.ServerConfig
```
Comment (crobert-1) [inline internal/receiver/signalfxgatewayprometheusremotewritereceiver/server.go:47]: Not a problem at all, but curious as to why this line was moved to the end of the struct?

**Review by crobert-1 — APPROVED** [review]

**Comment by crobert-1** [comment]
Thanks for all your work here @jeffreyc-splunk!

File: internal/receiver/signalfxgatewayprometheusremotewritereceiver/server.go (hunk 1 above)
Comment (jeffreyc-splunk) [reply internal/receiver/signalfxgatewayprometheusremotewritereceiver/server.go:47]: > Not a problem at all, but curious as to why this line was moved to the end of the struct?

Not sure. It was just the output from `make generate-metrics`: https://github.com/signalfx/splunk-otel-collector/pull/5001/commits/bc8fa8ce02010c2941f18c74580c2be461b895bb

**Comment by crobert-1** [comment]
Looks good to me. Looks like the main changes were some simple renames:
```
otelcol.NewCommand -> otelcol.NewCommandMustProviderSettings
CreateSettings -> Settings
```
And then a bit more of a complicated one:
```
UnmarshalConfig(conf *confmap.Conf, intoCfg Config) -> conf.Unmarshal(&intoCfg)
```
I didn't check every instance of the last change, but they all looked good from what I could see :+1:

**Pull request merged**